pkg/cmd/roachprod/main.go

```diff
@@ -121,26 +123,31 @@ func sortedClusters() []string {
 	return r
 }
 
-func newCluster(name string) (*install.SyncedCluster, error) {
-	nodeNames := "all"
-	{
-		parts := strings.Split(name, ":")
-		switch len(parts) {
-		case 2:
-			nodeNames = parts[1]
-			fallthrough
-		case 1:
-			name = parts[0]
-		case 0:
-			return nil, fmt.Errorf("no cluster specified")
-		default:
-			return nil, fmt.Errorf("invalid cluster name: %s", name)
-		}
+func parseClusterName(name string) (string, string, error) {
```

Comment:
Name the return arguments. Also, what's "name"? how about clusterName for the cluster name return type? And parseNodesAndClusterName for this helper + s/name/arg for the input?